# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/gl/GLContextEGL.h
# Commit: edfb73b96d5a
# Full Hash: edfb73b96d5ad946bbdd9da2e98d07b1f4565e1f
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2019-06-15 21:44:24
# Regressor Bug: 1559524
# File Overlap Count: 2
# Description:
#   Bug 1559524 - Reduce calls to GLLibraryEGL::Get(). r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D35103
# ==============================================================================

diff -r 0378898329b2 -r edfb73b96d5a gfx/gl/GLContextEGL.h
--- a/gfx/gl/GLContextEGL.h	Sat Jun 15 05:14:53 2019 +0000
+++ b/gfx/gl/GLContextEGL.h	Sat Jun 15 05:21:12 2019 +0000
@@ -21,12 +21,13 @@
   friend class TextureImageEGL;
 
   static already_AddRefed<GLContextEGL> CreateGLContext(
-      CreateContextFlags flags, const SurfaceCaps& caps, bool isOffscreen,
-      EGLConfig config, EGLSurface surface, nsACString* const out_failureId);
+      GLLibraryEGL*, CreateContextFlags flags, const SurfaceCaps& caps,
+      bool isOffscreen, EGLConfig config, EGLSurface surface,
+      nsACString* const out_failureId);
 
  public:
   MOZ_DECLARE_REFCOUNTED_VIRTUAL_TYPENAME(GLContextEGL, override)
-  GLContextEGL(CreateContextFlags flags, const SurfaceCaps& caps,
+  GLContextEGL(GLLibraryEGL*, CreateContextFlags flags, const SurfaceCaps& caps,
                bool isOffscreen, EGLConfig config, EGLSurface surface,
                EGLContext context);
 
@@ -47,11 +48,8 @@
 
   void SetIsDoubleBuffered(bool aIsDB) { mIsDoubleBuffered = aIsDB; }
 
-  virtual bool IsANGLE() const override {
-    return GLLibraryEGL::Get()->IsANGLE();
-  }
-
-  virtual bool IsWARP() const override { return GLLibraryEGL::Get()->IsWARP(); }
+  virtual bool IsANGLE() const override { return mEgl->IsANGLE(); }
+  virtual bool IsWARP() const override { return mEgl->IsWARP(); }
 
   virtual bool BindTexImage() override;
 
@@ -80,8 +78,6 @@
 
   EGLSurface GetEGLSurface() const { return mSurface; }
 
-  EGLDisplay GetEGLDisplay() const { return GLLibraryEGL::Get()->Display(); }
-
   bool BindTex2DOffscreen(GLContext* aOffscreen);
   void UnbindTex2DOffscreen(GLContext* aOffscreen);
   void BindOffscreenFramebuffer();
@@ -103,17 +99,14 @@
   virtual void OnMarkDestroyed() override;
 
  public:
+  const RefPtr<GLLibraryEGL> mEgl;
   const EGLConfig mConfig;
+  const EGLContext mContext;
 
  protected:
-  const RefPtr<GLLibraryEGL> mEgl;
   EGLSurface mSurface;
   const EGLSurface mFallbackSurface;
 
- public:
-  const EGLContext mContext;
-
- protected:
   EGLSurface mSurfaceOverride = EGL_NO_SURFACE;
   RefPtr<gfxASurface> mThebesSurface;
   bool mBound = false;
@@ -125,9 +118,10 @@
   bool mOwnsContext = true;
 
   static EGLSurface CreatePBufferSurfaceTryingPowerOfTwo(
-      EGLConfig config, EGLenum bindToTextureFormat, gfx::IntSize& pbsize);
+      GLLibraryEGL*, EGLConfig config, EGLenum bindToTextureFormat,
+      gfx::IntSize& pbsize);
 #if defined(MOZ_WAYLAND)
-  static EGLSurface CreateWaylandBufferSurface(EGLConfig config,
+  static EGLSurface CreateWaylandBufferSurface(GLLibraryEGL*, EGLConfig config,
                                                gfx::IntSize& pbsize);
 #endif
 #if defined(MOZ_WIDGET_ANDROID)
@@ -136,7 +130,8 @@
 #endif  // defined(MOZ_WIDGET_ANDROID)
 };
 
-bool CreateConfig(EGLConfig* config, int32_t depth, bool enableDepthBuffer);
+bool CreateConfig(GLLibraryEGL*, EGLConfig* config, int32_t depth,
+                  bool enableDepthBuffer);
 
 }  // namespace gl
 }  // namespace mozilla